### PR TITLE
feat: Added support for handler paths with hidden directories

### LINKED ISSUE #95: New Relic handler throws error when wrapping handler in hidden file or directory (node.js)
When the new relic handler tries to execute a wrapped function in a hidden directory it throws an error.

## Description

We are using the serverless-plugin-warmup to help minimize lambda cold starts. This plugin creates a hidden handler with the path `.warmup/default/index.warmUp`. 

https://github.com/newrelic/newrelic-lambda-layers/blob/379e55482c90eaa3af8258d94569f7a16ea2e361/nodejs/index.js#L24-L30

When the check on the above line runs, it detects that there are two periods in the path and then throws the error. It is techincally valid for a file path to contain multiple periods. as long as one is at the end and all others are prefixed with a forward slash. I believe a better check would be a regex similar to the following:

`^(\.?[\w\-]+\/)*\.?[\w\-]+\.\w+$`

## Steps to Reproduce

Create a new relic wrapped function with a handler that uses a hidden folder or file

## Expected Behaviour

The function executes

## Relevant Logs / Console output

```
2022-07-07T14:10:41.265Z	undefined	ERROR	Uncaught Exception 	
{
    "errorType": "Error",
    "errorMessage": "Improperly formatted handler environment variable: .warmup/default/index.warmUp",
    "stack": [
        "Error: Improperly formatted handler environment variable: .warmup/default/index.warmUp",
        "    at getHandler (/opt/nodejs/node_modules/newrelic-lambda-wrapper/index.js:27:11)",
        "    at Object.<anonymous> (/opt/nodejs/node_modules/newrelic-lambda-wrapper/index.js:62:50)",
        "    at Module._compile (internal/modules/cjs/loader.js:1085:14)",
        "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)",
        "    at Module.load (internal/modules/cjs/loader.js:950:32)",
        "    at Function.Module._load (internal/modules/cjs/loader.js:790:12)",
        "    at Module.require (internal/modules/cjs/loader.js:974:19)",
        "    at require (internal/modules/cjs/helpers.js:101:18)",
        "    at _tryRequire (/var/runtime/UserFunction.js:194:10)",
        "    at _loadUserApp (/var/runtime/UserFunction.js:215:12)"
    ]
}
```

## Your Environment

Runtime: Node.js 14.x

### PR DESCRIPTION
**Pull request by mrickard**
Closes #95 

Signed-off-by: mrickard <maurice@mauricerickard.com>

### CODE REVIEW

File: nodejs/index.js [hunk 1]
```diff
@@ -23,14 +23,20 @@ function getHandler() {
 
   const parts = handler.split('.')
 
-  if (parts.length !== 2) {
+  let moduleToImport
+  let handlerToWrap
+
+  if (parts.length === 2) {
```
Comment (bizob2828) [inline nodejs/index.js:29]: can't you just keep the new lastIndexOf logic for all?

You could do something like this

```
const lastDotIndex = handler.lastIndexOf('.')
const moduleToImport = handler.slice(0, lastDotIndex)
const handlerToWrap = handler.slice(lastDotIndex + 1)
```

**Review by bizob2828 — COMMENTED** [review]
I had a comment about solving this with 1 approach for all use cases.  Also, I'm not sure the format of this repo but I also noticed you didn't link the issue this is fixing.

File: nodejs/index.js (hunk 1 above)
Comment (mrickard) [reply nodejs/index.js:29]: @bizob2828 True--that would indeed apply to any length > 1. 

I did want to hold off on linking the issue until this was in full PR--I want to test first with a locally-built layer and an application with some dot directories.

Comment (bizob2828) [reply nodejs/index.js:29]: got it.  was there a reason why this doesn't have tests? It would be really easy to unit test this

Comment (mrickard) [reply nodejs/index.js:29]: The lack of tests came down to team size vs. the size of the docket of work. Not so much an explicit decision not to use tests, but more a lack of a decision to use them.

**Comment by mrickard** [comment]
> I had a comment about solving this with 1 approach for all use cases. Also, I'm not sure the format of this repo but I also noticed you didn't link the issue this is fixing.

Updated as a full PR with the issue linked.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**